src/lang/ir_operator.cc

```diff
@@ -258,7 +259,11 @@ Expr select(Expr cond, Expr true_value, Expr false_value) {
       return false_value;
     }
   }
-  return ir::Select::make(cond, true_value, false_value);
+  return ir::Call::make(
+      true_value.type(),
+      ir::intrinsic::tvm_if_then_else,
+      {cond, true_value, false_value},
+      ir::Call::Intrinsic);
```

Comment:
Why is it `Intrinsic` and not `PureIntrinsic`?

Reply:
Good point, I think we can treat if_then_else as pure as well